beater/api/mux.go

```diff
@@ -152,7 +143,7 @@ func sourcemapHandler(cfg *config.Config, builder *authorization.Builder, report
 	if err != nil {
 		return nil, err
 	}
-	h := sourcemap.Handler(systemMetadataDecoder(cfg, decoder.DecodeSourcemapFormData), psourcemap.Processor, *tcfg, reporter)
+	h := sourcemap.Handler(sourcemap.DecodeSourcemapFormData, psourcemap.Processor, *tcfg, reporter)
```

Comment:
why pass `sourcemap.DecodeSourcemapFormData` as an argument to the handler?
it would be simpler to unexport it, and call it directly from the handler in https://github.com/elastic/apm-server/blob/58ef039ae47c897c863429e6eaf9f2f3670209dd/beater/api/asset/sourcemap/handler.go#L55-L54

`ReqDecoder` would not be needed either.

Reply:
I personally agree, but I wanted to leave that change (which I thought might be contentious) to another PR. Doing it in this PR would require non-trivial changes to the tests, which currently rely on not having to know the details of the decoding.